src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { TelemetryService } from 'services';
+import { DeviceIcon } from './deviceIcon';
+import { RulesGrid, rulesColumnDefs } from 'components/pages/rules/rulesGrid';
+import { translateColumnDefs } from 'utilities';
+import {
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  FlyoutSection,
+  SectionHeader,
+  SectionDesc,
+  BasicGrid as Grid,
+  BasicRow as Row,
+  BasicHeaderCell as HeaderCell,
+  BasicCell as Cell
+} from 'components/shared';
+
+import './deviceDetails.css';
+
+export class DeviceDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { alarms: undefined, isAlarmsPending: false, alarmsError: undefined };
+    this.columnDefs = [
+      rulesColumnDefs.ruleName,
+      rulesColumnDefs.severity,
+      rulesColumnDefs.explore
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.fetchAlarms((nextProps.device || {}).id)
```

Comment:
I think you should check the device Ids here. Not totally sure but I think if you click the same device row twice it will still cause the props to update here. If the device ids are the same no need to call fetchAlarms again.